ui/hooks/multi-srp/useHdKeyringsWithSnapAccounts.ts

```diff
@@ -18,11 +18,13 @@ export const useHdKeyringsWithSnapAccounts = () => {
     getMetaMaskHdKeyrings,
   );
   const internalAccounts = useSelector(getInternalAccounts);
+
   return useMemo(() => {
     return hdKeyrings.map((keyring) => {
       const firstPartySnapAccounts = internalAccounts
         .filter(
           (account: InternalAccount) =>
+            account.metadata.snap?.id &&
```

Comment:
IMO, we should also check for `isSnapPreinstalled` here: https://github.com/MetaMask/metamask-extension/blob/6f29663a2feba884b477014d30c453eaafeeed16/shared/lib/snaps/snaps.ts#L19-L21

Given that we're looking for 1st party Snaps, they have to be preinstalled.